packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/utils.ts

```diff
@@ -0,0 +1,127 @@
+import { UniqueIdentifier } from "@dnd-kit/core";
+import { arrayMove } from "@dnd-kit/sortable";
+import { groupBy } from "lodash";
+import { AccountPage } from "../../../../components/hooks/useAccountPages";
+
+export interface PageList {
+  treeList: TreeElement[];
+  flatList: TreeElement[];
+}
+
+export interface TreeElement extends AccountPage {
+  parentId: string;
+  depth: number;
+  index: number;
+  expanded: boolean;
```

Comment:
This being stored here seems problematic to me - if you ever wanted to optimize this component it would get very difficult as you have to redo this entire tree calculation every time you open/close a page. Can you look at moving this out of the page state?

Reply:
Moved all collapse/expand state calculations to the page item's render function in 5cab3ac1a6df55c19655ea81a335bce2e28caeac